src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/DefaultODataDeserializerProvider.cs

```diff
@@ -88,6 +88,13 @@ internal ODataDeserializer GetODataDeserializerImpl(Type type, Func<IEdmModel> m
                 return _rootContainer.GetRequiredService<ODataActionPayloadDeserializer>();
             }
 
+            if (type == typeof(EdmChangedObjectCollection) ||
+               (type.IsGenericType && type.GetGenericTypeDefinition() == typeof(DeltaSet<>)))
+            {
+                return _rootContainer.GetRequiredService<ODataResourceSetDeserializer>();
+            }
+
```

Comment:
actually, these part should move to "GetEdmType(...)" method.
ODataResoureSetDeserializer should return from GetEdmTypeDeserializer(edmType).